remix/app/routes/_app.newsletter.update.$id.test.tsx

```diff
@@ -35,65 +36,112 @@ describe('PUT /newsletter/update/:id', () => {
+        prismaMock.newsletter.findFirst.mockResolvedValue(newsletter);
 
         const request = makeRequest({
           url: `/newsletter/update/${newsletter.id}`,
-          method: 'put',
-          body,
+          method: 'get',
         });
 
-        const result: any = await action({
+        const result: any = await loader({
           request,
           params: {},
           context: {},
         });
 
-        expect(await result.json()).toMatchObject({
+        expect(await result.json()).toMatchObject<Newsletter>({
+          id: newsletter.id,
           name: expect.any(String),
           content: expect.any(String),
+          userId: newsletter.userId,
+          createAt: expect.any(String),
+          updateAt: expect.any(String),
         });
       });
     });
+  });
 
-    describe('given an invalid newsletter params', () => {
-      it('returns error NOT FOUND', async () => {
-        const newsletter = { ...newsletterFactory };
+  describe('action', () => {
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
 
-        const body = new URLSearchParams({
-          name: newsletter.name,
-          content: newsletter.content,
-        });
+    describe('given the user is authenticated', () => {
+      const userAttributes = { id: '1' };
+      const user = { ...userFactory, ...userAttributes };
+
+      (appHandler as jest.Mock).mockImplementation((req, callback) =>
+        callback(user)
+      );
+
+      describe('given valid newsletter params', () => {
+        it('update a newsletter', async () => {
+          const updateResultMock = { count: 1 };
+          const newsletterAttributes = { userId: '1' };
+          const newsletter = {
+            ...newsletterFactory,
+            ...newsletterAttributes,
+          };
+
+          prismaMock.newsletter.findUnique.mockResolvedValue(newsletter);
+          prismaMock.newsletter.updateMany.mockResolvedValue(updateResultMock);
+
+          const body = new URLSearchParams({
+            name: newsletter.name,
+            content: newsletter.content,
+          });
 
-        const request = makeRequest({
-          url: `/newsletter/update/${newsletter.id}`,
-          method: 'put',
-          body,
-        });
+          const request = makeRequest({
+            url: `/newsletter/update/${newsletter.id}`,
+            method: 'put',
+            body,
+          });
 
-        try {
-          await action({
+          const result: any = await action({
             request,
             params: {},
             context: {},
           });
-        } catch (error) {
-          expect(error).toMatchObject({
-            status: StatusCodes.NOT_FOUND,
-            statusText: 'Newsletter not found',
+
+          expect(await result.json()).toMatchObject({
+            name: expect.any(String),
+            content: expect.any(String),
           });
-        }
+        });
+      });
+
+      describe('given an invalid newsletter params', () => {
```

Comment:
Syntax issue:

```suggestion
      describe('given invalid newsletter params', () => {
```

Since it is plural, we cannot use `an`

Reply:
Yep Fixed here: [b357dd3](https://github.com/nimblehq/js-framework-benchmark/pull/130/commits/b357dd39fc2364fe1acf62523fa9f7463e88caed)